### PR TITLE
make the fieldsets have a wrapper with a header and body

### PR DESCRIPTION
**Pull request by hippogriffic**
so the fields for each fieldset can be hidden / shown with some ng-magic to be added later. Max is responsible for 99% of this, I barely know what's going on at this point. But it works so yay.

@incuna/frontend please review
(the tests probably need fixing)

part of https://incuna-pm.atlassian.net/browse/OP-170

### CODE REVIEW

File: web/templates/twig-source/incuna-surveys/fields/wrapper.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% raw %}
+
+        <div class="panel panel-primary">
+
+          <div class="panel-heading px-nested-panel-heading clearfix" id="id-{{options.templateOptions.id}}">
+               <h3 ng-bind="options.templateOptions.name" class="field-group-name"></h3>
+               <h4 ng-bind="options.templateOptions.description" class="field-group-desc"></h4>
+          </div>
+test
```
Comment (wytrych) [inline web/templates/twig-source/incuna-surveys/fields/wrapper.html]: Remove test

File: web/templates/twig-source/incuna-surveys/fields/wrapper.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+          <div class="panel-body px-nested-panel-body">
+            <formly-transclude></formly-transclude>
```
Comment (wytrych) [inline web/templates/twig-source/incuna-surveys/fields/wrapper.html]: Fix indentation

File: web/app/scripts/src/services/fieldsets-parser.es.js [hunk 3]
```diff
@@ -25,24 +25,14 @@ module.service(moduleProperties.componentName, [
...
-                        templateOptions: {
-                            fieldGroupName: fieldset.name,
-                            fieldGroupDesc: fieldset.description
-                        }
+                        wrapper: 'panel',
+                        templateOptions: fieldset,
+                        fieldGroup: []
                     }
-                ];
+                ;
```
Comment (wytrych) [inline web/app/scripts/src/services/fieldsets-parser.es.js]: This comma is in wrong place and fix indentation

File: web/app/scripts/src/services/fieldsets-parser.es.js [hunk 4]
```diff
@@ -62,12 +52,12 @@ module.service(moduleProperties.componentName, [
...
 
-                    fieldGroup.push(fieldObject);
+                    fieldGroup.fieldGroup.push(fieldObject);
                 });
 
-                fields.push({
+                fields.push(
                     fieldGroup
-                });
+                );
```
Comment (wytrych) [inline web/app/scripts/src/services/fieldsets-parser.es.js]: Make this one line

File: web/app/scripts/src/services/fieldsets-parser.es.js [hunk 5]
```diff
@@ -25,24 +25,14 @@ module.service(moduleProperties.componentName, [
...
-                templateUrl: FieldsConfig.headerTemplateUrl,
-                templateOptions: {
-                    formName: form.name,
-                    formDescription: form.description
-                }
-            };
-
             form.fieldsets.forEach((fieldset) => {
-                let fieldGroup = [
+                let fieldGroup =
```
Comment (wytrych) [inline web/app/scripts/src/services/fieldsets-parser.es.js]: This should be called `fieldGroupWrapper` I think

**Comment by maxpeterson** [comment]
Looking at it now

**Comment by maxpeterson** [comment]
@incuna/js js is ready for review now. @hippogriffic will update the html shortly.

@wytrych the structure of the `fields` is now much simpler, just a list of fieldset (group) objects containing a list of fields. 
```
[
{
                    wrapper: 'panel',
                    templateOptions: fieldset,
                    fieldGroup: [
                        {
                            key: field.id,
                            type: field.field_type,
                            templateOptions: {
                                fieldSetId: fieldset.id,
                                autoId: `id-${fieldset.id}-${field.id}`,
                                choices: field.answers,
                                fieldOptions: {
                                    // jscs:disable disallowQuotedKeysInObjects
                                    'help_text': field.help_text,
                                    required: field.required,
                                    label: field.name
                                    // jscs:enable disallowQuotedKeysInObjects
                                }
                            }
                        },
                        ...
                    ]
                }
...
]
```
 
Likewise the html is now a list of fieldset / group divs containing fields.

File: web/app/scripts/src/services/fields-config.es.js [hunk 6]
```diff
@@ -12,8 +12,6 @@ const module = angular.module(moduleProperties.moduleName, [
 module.service(moduleProperties.componentName, [
     function () {
         this.templatesBase = 'templates/incuna-surveys/fields';
-        this.headerTemplateUrl = `${this.templatesBase}/header.html`;
```
Comment (maxpeterson) [inline web/app/scripts/src/services/fields-config.es.js]: The header is to the `surveys-form` directive.

File: web/app/scripts/src/services/fields-config.es.js [hunk 7]
```diff
@@ -12,8 +12,6 @@ const module = angular.module(moduleProperties.moduleName, [
...
-        this.fieldsetHeaderTemplateUrl = `${this.templatesBase}/fieldset-header.html`;
```
Comment (maxpeterson) [inline web/app/scripts/src/services/fields-config.es.js]: The fieldset now has it's own wrapper template that renders the title and fields.

File: web/templates/twig-source/incuna-surveys/form/survey-form.html [hunk 8]
```diff
@@ -1,4 +1,7 @@
 <form ng-submit="submit()">
+    <h1 ng-bind="form.name" class="form-name"></h1>
```
Comment (maxpeterson) [inline web/templates/twig-source/incuna-surveys/form/survey-form.html]: Form title moved to here from the `formly` form.

**Comment by wytrych** [comment]
Nice

**Pull request merged**